framework/src/st_things/things_stack/utils/things_hashmap.c

```diff
@@ -167,13 +167,11 @@ void *hashmap_get(struct hashmap_s *hash, unsigned long key)
 unsigned long* hashmap_get_keyset(struct hashmap_s *hash)
 {
 	unsigned long* keyset = NULL;
-	int idx = 0;
 
 	if (hash->count) {
-		int i;
-
+		int i = 0;
+		int idx = 0;
 		keyset = things_malloc(sizeof(unsigned long) * hash->count);
-
 		for (i = 0; i < hash->size; i++) {
 			if (hash->table[i].flags & ACTIVE) {
 
```

Comment:
i think idx value is needed ++

Reply:
Okay. 

Note: It's existing code and not reported by static analyzer.